File: app/src/main/java/com/example/daisyling/common/util/download/MainThreadExecutor.java
package com.example.daisyling.common.util.download;

import android.os.Handler;
import android.os.Looper;

import java.util.concurrent.Executor;

/**
 * Created by Emily on 10/11/21
 */
public class MainThreadExecutor implements Executor {
    private final Handler handler = new Handler(Looper.getMainLooper());

    @Override
    public void execute(Runnable r) {
        handler.post(r);
    }
}

File: app/src/main/java/com/example/daisyling/common/util/download/MainThreadExecutor.kt
package com.example.daisyling.common.util.download

import android.os.Handler
import android.os.Looper
import java.util.concurrent.Executor

/**
 * Created by Emily on 10/11/21
 */
class MainThreadExecutor : Executor {
    private val handler = Handler(Looper.getMainLooper())
    override fun execute(r: Runnable) {
        handler.post(r)
    }
}